### PR TITLE
[#5601] feat(core): Add model entity and model dispatcher in Gravitino

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

1. Define and add the model namespace, name identifier, entity definition in Gravitino.
2. Add the basic dispatcher framework in Gravitino.

### Why are the changes needed?

This is the second PR to support model management in Gravitino.

Fix: #5601 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Add UTs to verify part of the code.

### CODE REVIEW

**Comment by FANNG1** [comment]
LGTM

File: core/src/main/java/org/apache/gravitino/catalog/ModelNormalizeDispatcher.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+      return idents;
+    }
+
+    Capability capabilities = getCapability(idents[0], catalogManager);
+    return applyCaseSensitive(idents, Capability.Scope.MODEL, capabilities);
+  }
+
+  private NameIdentifier normalizeNameIdentifier(NameIdentifier ident) {
+    Capability capability = getCapability(ident, catalogManager);
+    return applyCapabilities(ident, Capability.Scope.TOPIC, capability);
```
Comment (yuqi1129) [inline core/src/main/java/org/apache/gravitino/catalog/ModelNormalizeDispatcher.java:165]: Why the scope is the `topic` here?

**Review by yuqi1129 — COMMENTED** [review]
Just a minor one, others sees to be good to me.

File: core/src/main/java/org/apache/gravitino/catalog/ModelNormalizeDispatcher.java (hunk 1 above)
Comment (jerryshao) [reply core/src/main/java/org/apache/gravitino/catalog/ModelNormalizeDispatcher.java:165]: Sorry, just a typo, I will fix it.

File: core/src/main/java/org/apache/gravitino/meta/ModelVersionEntity.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  public static final Field COMMENT =
+      Field.optional("comment", String.class, "The comment or description of the model entity.");
+  public static final Field ALIASES =
+      Field.optional("aliases", List.class, "The aliases of the model entity.");
+  public static final Field URL =
+      Field.required("uri", String.class, "The URI of the model entity.");
+  public static final Field PROPERTIES =
+      Field.optional("properties", Map.class, "The properties of the model entity.");
+  public static final Field AUDIT_INFO =
+      Field.required("audit_info", AuditInfo.class, "The audit details of the model entity.");
```
Comment (mchades) [inline core/src/main/java/org/apache/gravitino/meta/ModelVersionEntity.java:45]: The `ModelVersionEntity` doesn't include the model info?

Does `ModelVersionEntity` not need to include information about `Model`?

Comment (jerryshao) [reply core/src/main/java/org/apache/gravitino/meta/ModelVersionEntity.java:45]: In my current implementation, it is not needed?

Comment (mchades) [reply core/src/main/java/org/apache/gravitino/meta/ModelVersionEntity.java:45]: if so, how can we determine which `ModelVersionEntity` belongs to which `Model`?

**Review by mchades — APPROVED** [review]
LGTM.

**Review by yuqi1129 — APPROVED** [review]

**Pull request merged**